BillNye.py

```diff
@@ -0,0 +1,161 @@
+    pxarray = pygame.PixelArray(DISPLAY)
+    for y in xrange(HEIGHT):
+        for x in xrange(WIDTH):
+            RED = DISPLAY.get_at((x, y)).r
+            GREEN = DISPLAY.get_at((x, y)).g
+            BLUE = DISPLAY.get_at((x, y)).b
+            pxarray[x, y] = (255 - RED, GREEN, 255- BLUE)
+
+def blueColour():
+    DISPLAY.blit(colourImage, (0, 0))
+    pygame.display.flip()
+    pxarray = pygame.PixelArray(DISPLAY)
+    for y in xrange(HEIGHT):
+        for x in xrange(WIDTH):
+            RED = DISPLAY.get_at((x, y)).r
+            GREEN = DISPLAY.get_at((x, y)).g
+            BLUE = DISPLAY.get_at((x, y)).b
+            pxarray[x, y] = (255 - RED, 255 - GREEN, BLUE)
+
+def invertColour():
+    pxarray = pygame.PixelArray(DISPLAY)
+    for y in xrange(HEIGHT):
+        for x in xrange(WIDTH):
+            RED = DISPLAY.get_at((x, y)).r
+            GREEN = DISPLAY.get_at((x, y)).g
+            BLUE = DISPLAY.get_at((x, y)).b
+            pxarray[x, y] = (255 - RED, 255 - GREEN, 255 - BLUE)
+    del pxarray
+
+def BlackWhite():
+    DISPLAY.blit(colourImage, (0, 0))
+    pygame.display.flip()
+    pxarray = pygame.PixelArray(DISPLAY)
+    for y in xrange(HEIGHT):
+        for x in xrange(WIDTH):
+            RED = DISPLAY.get_at((x, y)).r
+            GREEN = DISPLAY.get_at((x, y)).g
+            BLUE = DISPLAY.get_at((x, y)).b
+
+            GREY = (RED + GREEN + BLUE)
+            if GREY < 400:
+                pxarray[x, y] = (0, 0, 0)
+            else:
+                pxarray[x, y] = (255, 255, 255)
+
+    del pxarray
+
+while True:
+   for event in pygame.event.get():
+       if event.type == QUIT:
+           pygame.quit()
+           sys.exit()
+   if event.type == pygame.KEYDOWN:
+       if event.key == pygame.K_1:
+           Greyscale()
+           pygame.display.flip()
+           pygame.image.save(DISPLAY, 'images\BillNyeGrey.jpg')
+   if event.type == pygame.KEYDOWN:
+       if event.key == pygame.K_2:
+           redColour()
+           invertColour()
+           pygame.display.flip()
+           pygame.image.save(DISPLAY, 'images\BillNyeRed.jpg')
+   if event.type == pygame.KEYDOWN:
+       if event.key == pygame.K_3:
+           greenColour()
+           invertColour()
+           pygame.display.flip()
+           pygame.image.save(DISPLAY, 'images\BillNyeGreen.jpg')
+   if event.type == pygame.KEYDOWN:
+       if event.key == pygame.K_4:
+           blueColour()
+           invertColour()
+           pygame.display.flip()
+           pygame.image.save(DISPLAY, 'images\BillNyeBlue.jpg')
+   if event.type == pygame.KEYDOWN:
+       if event.key == pygame.K_5:
+           BlackWhite()
+           pygame.display.flip()
+           pygame.image.save(DISPLAY, 'images\BillNyeBlackWhite.jpg')
+   if event.type == pygame.KEYDOWN:
+       RedImage = pygame.image.load('images\BillNyeRed.jpg')
+       RedImage = pygame.transform.scale(RedImage, (700, 466))
+       GreenImage = pygame.image.load('images\BillNyeGreen.jpg')

```

Comment:
If these files do not yet exist, they should be created automatically through a relevant function call.